visual-studio-vstsbuildagent-vm/InstallVSTSAgent.ps1

```diff
@@ -81,7 +83,88 @@ Write-Verbose "Configuring agent" -Verbose
 # Set the current directory to the agent dedicated one previously created.
 Push-Location -Path $agentInstallationPath
 
-.\config.cmd --unattended --url $serverUrl --auth PAT --token $PersonalAccessToken --pool $PoolName --agent $AgentName --runasservice
+if ($runAsAutoLogon -ieq "true")
+{
+  # Create a PS session for the user to trigger the creation of the registry entries required for autologon
+  $computerName = "localhost"
+  $password�=�ConvertTo-SecureString�$vmAdminPassword -AsPlainText�-Force
+  if ($vmAdminUserName.Split("\").Count -eq 2)
+  {
+    $domain = $vmAdminUserName.Split("\")[0]
+    $userName = $vmAdminUserName.Split('\')[1]
+  }
+  else
+  {
+    $domain = Hostname
+    $userName = $vmAdminUserName
+  }
+  $credentials =�New-Object�System.Management.Automation.PSCredential("$domain\\$userName", $password)
+  Enter-PSSession�-ComputerName�$computerName�-Credential�$credentials
```

Comment:
do you want ignore it or fail if this fails to connect?

Reply:
I'm thinking fail, but not very sure.

Reply:
It is better we fail here otherwise the agent will register itself to vsts but show up red because it fails to start on machine restart. You'll basically have a dead agent on vsts which the user had to manually remove because nothing short of reconfiguring the agent after rdping into the machine will bring it back up successfully